src/modm/driver/encoder/encoder_impl.hpp

```diff
@@ -2,27 +2,25 @@
 #error "Don't include this file directly, use 'encoder.hpp' instead!"
 #endif
 
-template<typename PH_A, typename PH_B, modm::encoder::Division_t DIVISION, typename UnderlyingType>
+template<typename PH_A, typename PH_B, modm::encoder::Division_t DIVISION, std::signed_integral UnderlyingType>
 inline void
 modm::Encoder<PH_A, PH_B, DIVISION, UnderlyingType>::update()
 {
-	// Tare on power-on
+	const auto read = Phases::read();
+	
 	UnderlyingType raw = 0;
-	UnderlyingType diff;
-
-	auto read = Phases::read();
 	if (read & 0b10) raw = 3;
 	if (read & 0b01) raw ^= 1;  // convert gray to binary
```

Comment:
I'm not sure, if this isn't already optimized by the compiler. Using a lookup-table instead of bit-ops?
```
	uint8_t raw_lookup[] = {0,1,3,2};
	UnderlyingType raw = raw_lookup[Phases::read()];
```

Reply:
Hard to say, I generally check with `scons listing` then searching for the function in the .lss file and comparing the generated assembly.

Reply:
Ok, let me go for this.

Reply:
Okay forget about it: Initializing the list comsumes lot's more ticks than the bit-ops.

I've measured the period of update() without signal changes using simple Pin::set() / pin::reset():

ATmega328p@16Mhz:
With BitOps: 1.5us
With Lookup: 2.312us

Reply:
Oh wow, I would have guessed the exact opposite!

Reply:
Double checked again... and jepp it's true.